### ISSUE TITLE
Feature Request: Add new feature to `keras.utils.pad_sequences` to include the truncated part as a new instance in dataset

### ISSUE THREAD

**Issue by VachanVY** [keras-team/keras#18869]
## Required behaviour:
New argument for required behaviour: `truncate_and_include`: `"pre"`, `"post"` or `None` (if not required)
```python
>>> import keras as nn
>>> maxlen = 7
>>> ds = [[1,2,3,4,5,6,7,8,9, 10, 11, 12], [13,14,15,16,17,18]]
>>> nn.utils.pad_sequences(
        ds, 
        maxlen=maxlen,
        padding="post",
        truncate_and_include="post", # new argument for required behaviour
    )
array([[1, 2, 3, 4, 5, 6, 7], 
       [8, 9, 10, 11, 12, 0, 0],
       [13,14,15,16,17,18,0,]])
```
As you can see in the example it didn't exclude the truncated part but included the truncated as a new instance with padding if required.
It'll be also be helpful to include an argument to completely exclude sequences longer than maxlen

**Comment by haifeng-jin** [comment]
Hi @VachanVY , we recommend using `keras.ops`, which is a set of numpy operations to implement this feature.

And `pad_sequences` only works with numpy arrays, not cross backend.
We put this API out mainly for backward compatibility reasons.

**Comment by VachanVY** [comment]
Hi @haifeng-jin, this feature is frequently used in various NLP tasks like summarization, Q&A, etc. May be useful if his feature is included...
